interfaces/builtin/serial_port.go

```diff
@@ -208,6 +210,64 @@ func (iface *serialPortInterface) AutoConnect(*snap.PlugInfo, *snap.SlotInfo) bo
 	return true
 }
 
+func (iface *serialPortInterface) HotplugDeviceDetected(di *hotplug.HotplugDeviceInfo) (*hotplug.ProposedSlot, error) {
+	bus, _ := di.Attribute("ID_BUS")
+	if di.Subsystem() != "tty" || bus != "usb" || !serialDeviceNodePattern.MatchString(di.DeviceName()) {
+		return nil, nil
+	}
+
+	slot := hotplug.ProposedSlot{
+		Attrs: map[string]interface{}{
+			"path": di.DeviceName(),
+		},
+	}
+	if vendor, ok := di.Attribute("ID_VENDOR_ID"); ok {
+		slot.Attrs["usb-vendor"] = vendor
```

Comment:
I don't know if it is too late to change that, can the attribute be called `usb-vendor-id`, to differentiate it from any (variable and optional) vendor *name*?

Reply:
Yes, it's too late, this cannot be changed, these are existing attributes that come from gadget yaml if serial port is defined there.